### PR TITLE
Read gobee config from home

### PR DESCRIPTION
**Pull request by mattcanty**

### CODE REVIEW

**Comment by mattcanty** [comment]
@Kyslik not sure how to add you as a reviewer. Never collaborated on my own repos before!

**Comment by Kyslik** [comment]
I can't add myself either. I guess I would need to be a collaborator or contributor 🤷‍♂ .

**Comment by mattcanty** [comment]
> I can't add myself either. I guess I would need to be a collaborator or contributor 🤷‍♂ .

Would you like to be a collaborator? Happy to add you.

**Comment by Kyslik** [comment]
Sure I have no problem with that.

**Comment by mattcanty** [comment]
Consider yourself invited

File: main.go [hunk 1]
```diff
@@ -13,18 +12,14 @@ import (
...
-		fmt.Println("Please specify your config file using the -f option")
-		return
+	home, err := os.UserHomeDir()
+	if err != nil {
+		log.Fatal(err)
 	}
-
-	f, err := os.Open(fileName)
+	defaultConfigPath := fmt.Sprintf("%s/.config/gobee/config.yaml", home)
+	f, err := os.Open(defaultConfigPath)
```
Comment (Kyslik) [inline main.go]: You could use `ioutil.ReadFile()` and get slice of bytes (`byte[]`) and later on `yaml.Unmarshal(contents, target)`. Just an alternative current way is fine as well.

Comment (mattcanty) [reply main.go]: `os.Open()` appears more succinct, is there more overhead with this function than `ioutil.ReadFile()`?

Comment (mattcanty) [reply main.go]: Or does it mean that I would swap
```
	decoder := yaml.NewDecoder(f)
	err = decoder.Decode(&cfg)
```
for
```
yaml.Unmarshal(contents, target)
```
?

Comment (Kyslik) [reply main.go]: From code I read `os.Open` is mostly used if there is more things to do with the file. `ioutil.ReadFile` is wrapper around it and less verbose in the end.

Comment (Kyslik) [reply main.go]: Something like:

```go
	configContents, err := ioutil.ReadFile(defaultConfigPath)
	if err != nil {
		log.Fatalf("%s\nHave you created a Gobee config file at %s?", err, defaultConfigPath)
	}

	var cfg config.Config
	err = yaml.Unmarshal(configContents, &cfg)
	if err != nil {
		log.Fatalf("cmd.Run() failed with %s\n", err)
	}
```

Comment (mattcanty) [reply main.go]: Like it. 
Done.

**Review by Kyslik — APPROVED** [review]

File: main.go (hunk 1 above)
Comment (Kyslik) [reply main.go]: The config is not going to be megabytes long no need to stream it; that's another reason.

**Comment by mattcanty** [comment]
this https://github.community/t5/How-to-use-Git-and-GitHub/Github-projects-issues-and-pull-requests/td-p/12367

🤦‍♂

**Pull request merged**